src/DoubleShapeletPsfApprox.cc

```diff
@@ -283,22 +283,24 @@ class ProfileObjective : public OptimizerObjective {
         Scalar oAlpha = parameters[1] * _normalization;
         Scalar iR2 = parameters[2] * parameters[2];
         Scalar oR2 = parameters[3] * parameters[3];
-        residuals.asEigen<Eigen::ArrayXpr>() = _data
-            - (iAlpha/iR2)*(_arg/iR2).exp()
-            - (oAlpha/oR2)*(_arg/oR2).exp();
+        auto argEigen = _arg.asEigen<Eigen::ArrayXpr>();
+        residuals.asEigen<Eigen::ArrayXpr>() = _data.asEigen<Eigen::ArrayXpr>()
+            - (iAlpha/iR2)*(argEigen/iR2).exp()
+            - (oAlpha/oR2)*(argEigen/oR2).exp();
     }
 
     virtual bool differentiateResiduals(
         ndarray::Array<Scalar const,1,1> const & parameters,
         ndarray::Array<Scalar,2,-2> const & derivatives
     ) const {
-        ndarray::EigenView<Scalar,2,-2,Eigen::ArrayXpr> d(derivatives);
+        auto d = derivatives.asEigen<Eigen::ArrayXpr>();
+        auto argEigen = _arg.asEigen<Eigen::ArrayXpr>();
```

Comment:
You end up doing these conversions a lot. Consider using private getters or simply letting the types of `_data` and `_arg` break when you switch to `Eigen::Map`.

Reply:
The plan is as follows: add `asEigenArray` and `asEigenMatrix` to `ndarray::Array`; these will both return Eigen::Map. Then eliminate all use of `asEigen` in our code. This will eliminate all use of ndarray::EigenView, which can then go away (at least conditionally when ndarray is built against too new a version of Eigen, but as far as LSST is concerned it will go away).

Reply:
How does that reduce the code duplication of translating `_data` and `_arg` to `Eigen::Map` every time you want to use them?